src/services/authorization.ts

```diff
@@ -26,27 +28,120 @@ const permissionMapping = {
  * @param item
  * @throws if the user cannot access the item
  */
+
+export const validatePermissionMany = async (
+  {
+    itemMembershipRepository,
+    itemTagRepository,
+  }: {
+    itemMembershipRepository: typeof ItemMembershipRepository;
+    itemTagRepository: typeof ItemTagRepository;
+  },
+  permission: PermissionLevel,
+  member: Actor,
+  items: Item[],
+): Promise<ResultOf<ItemMembership | null>> => {
+  // batch request for all items
+  const inheritedMemberships = member
+    ? await itemMembershipRepository.getInheritedMany(items, member, true)
+    : null;
+  const tags = await itemTagRepository.hasManyForMany(items, [
+    ItemTagType.Public,
+    ItemTagType.Hidden,
+  ]);
+
+  const result: ResultOf<ItemMembership | null> = {
+    data: inheritedMemberships?.data ?? {},
+    errors: [],
+  };
+
+  for (const item of items) {
+    const highest = result.data[item.id]?.permission;
+    const isValid = highest && permissionMapping[highest].includes(permission);
```

Comment:
Is this equivalent to (i.e. can you replace instances of `permissionMapping` with):
```suggestion
    const isValid = highest && PermissionLevelCompare.lte(permission, highest);
```

Reply:
I don't know how this part work exactly, I just copied it from the "validateOne" implementation

Reply:
In fact that might be true, but rather:
```suggestion
 const isValid = highest && PermissionLevelCompare.gte(permission, highest);
```